node/test/api/tag.js

```diff
@@ -168,4 +168,36 @@ describe('Tag API', () => {
         assert(response.body, "tag deleted from user");
     });
 
+    it('should return 5 tags', async () => {
+        // create 5 tags
+        for(let i = 0; i < 5; i++){
+            await factories.tag();
+        }
+        //make API call
+        const response = await chai.request(app).get('/api/tags/recommend')
+        expect(response.body).to.have.lengthOf(5)
+    });
+
+    it('should return no more than 5 tags', async () => {
+        // create 5 tags
+        for(let i = 0; i < 7; i++){
+            await factories.tag();
+        }
+        //make API call
+        const response = await chai.request(app).get('/api/tags/recommend')
+        expect(response.body).to.have.lengthOf(5)

```

Comment:
Since this is a “no more than 5” option, wouldn’t this fail if there are only 4 tags?

Reply:
I was thinking the api call should return up to 5 tags, so if only 4 tags exist in the database, then only 4 would be returned, but you can't be more than 5.

Reply:
In this case if this only returned 4 tags this testcase would fail. because there is suppose to be 7 tags in the database

Reply:
Ah I see, I understand now, seems good to go then